### DISCUSSION TITLE
A prototypical set of `rust-bitcoin` users

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#2198] [General]
In order to write a library that is correct, usable, and ergonomic it would help to know how our users are using the lib. We can't look at every user but we can come up with a set of users, that when taken together, cover most usage patterns of `rust-bitcoin`. Considering this set of users at release time would also help us test release candidates.

Proposed set of users (in no particular order):

- `rust-lighting`
- `bdk`
- `electrs`
- `rust-miniscript`

EDIT: Proposed acronym so we can refer to this set: PSU

**Comment by tcharding** [comment]
Originally discussed in: https://github.com/rust-bitcoin/rust-bitcoin/pull/2152#issuecomment-1793120196

**Comment by tcharding** [comment]
@dpc is it fair to say that Fedimint is not leveraging `rust-bitcoin` in any way not covered by `bdk` and `rust-lightning`?

**Reply by dpc** [reply]
Hard for me to say, but I'd guess that overlap with the projects you already listed should be large.

**Comment by Kixunil** [comment]
We should really look at what APIs they use. I suspect your list is likely missing advanced Taproot APIs, maybe even block filters or other stuff. For advanced Taproot we can use Firefish (not yet open, I hope it'll be soon; but I still can provide insight on it as I already did).
